packages/react-async/README.md

```diff
@@ -0,0 +1,147 @@
+  renderPrefetch: () => <MyQuery.Prefetch />,
+  renderPreload: () => <MyQuery.Preload />,
+  renderKeepFresh: () => <MyQuery.KeepFresh />,
+});
+```
+
+### `Prepare`
+
+The `Prepare` component allows you to use the asynchronous component you generated with `createAsyncComponent` and automatically render its `Prefetch` component when the user looks like they are going to navigate to a page that uses it. This component takes as its props the asynchronous component, a URL pattern to look for (a string or `RegExp` that is compared against the target pathname), and an optional function that can map the URL to a set of props for your prefetch component.
+
+Consider this async component:
+
+```tsx
+const ProductDetails = createAsyncComponent({
+  load: () => import('./ProductDetails'),
+  renderPrefetch: ({id}: {id: string}) => <PrefetchGraphQLQuery id={id} />,
+});
+```
+
+This component might be rendered when the URL matches `/products/:id`. If we want to prefetch this component (including its GraphQL query!) whenever the user is going to navigate to a matching URL, we would register this intent with the following `Prepare` component:
+
+```tsx
+<Prefetch
+  component={ProductDetails}
+  url={/^\/products\/(\d+)$/}
+  mapUrlToProps={pathname => {
+    // If you don’t pass the right types, or don’t pass mapUrlToProps()
+    // at all, you’ll get a type error.
+    const id = pathname.split('/').pop();
+    return {id};
+  }}
+/>
+```
+
+And that’s it. While we reserve the right to change it, the basic process for determining merchant navigation intent is fairly simple. We listen for users mousing over or focusing in to elements with an `href` attribute (or, `data-href`, if you can’t use a real link) and, if the user doesn’t mouse/ focus out in some small amount of time, we prefetch all matching components. We also do the prefetch when the user begins their click on an element with an `href` attribute.
+
+### `AsyncManager` and `AsyncProvider`
+
+`AsyncManager` and `AsyncProvider` allow you to extract the asynchronous bundles that were required for your application. If you use the Babel plugin, every component created by `createAsyncComponent` will report its existence when rendered to an `AsyncManager`. So, on the server, you simply need to provide an `AsyncManager` to your app, render the app to a string, and then inspect the `AsyncManager`’s `used` property.
+
+```tsx
+import {AsyncManager, AsyncProvider} from '@shopify/react-async';
+
+const asyncManager = new AsyncManager();
+
+renderToString(
+  <AsyncProvider manager={asyncManager}>
+    <App />
+  </AsyncProvider>,
+);
+
+const moduleIds = [...asyncManager.used];
+```
+
+These module IDs can be looked up in the manifest created by `@shopify/async`’s Webpack plugin. If you are using [`sewing-kit-koa`](https://github.com/Shopify/quilt/tree/master/packages/sewing-kit-koa), you can follow the instructions from that package to automatically collect the required JavaScript and CSS bundles.
```

Comment:
> These module IDs can be looked up in the manifest created by `@shopify/async`’s Webpack plugin.

Might be worth including the default manifest path `build/client/async-manifest.json`.

Reply:
The manifest path depends on Rails versus Node, and is not something we otherwise mention since `sewing-kit-koa` handles it internally